src/services/asset.ts

```diff
@@ -1,12 +1,38 @@
-import { S3 } from 'aws-sdk';
+import { S3 } from "aws-sdk";
 import { configManager } from "./config-manager";
 
-export default (s3: S3, config: typeof configManager) => ({
-  getAssetUrl: (articleId: string, fileKey: string): string => {
-    return s3.getSignedUrl('getObject', {
-        Bucket: config.get('editorS3Bucket'),
-        Key: `${articleId}/${fileKey}`,
-        Expires: 3600
-    })
+async function checkFileExists(s3: S3, key: string, bucket: string) {
+  try {
+    await s3
+      .headObject({
+        Bucket: bucket,
+        Key: key,
+      })
+      .promise();
+    return true;
+  } catch (error) {
+    if (error && error.code === "NotFound") {
+      return false;
+    }
+    throw error;
   }
-});
\ No newline at end of file
+}
+
+export default (s3: S3, config: typeof configManager) => ({
+  getAssetUrl: async (
```

Comment:
I'm not a big fan of default exports. Having a name here would really explain what this function does. I can only assume it returns a string which is a signed URL.

Reply:
Addressed - named assetService